store/v2/commitment/mem/tree.go

```diff
@@ -0,0 +1,67 @@
+package mem
+
+import (
+	ics23 "github.com/cosmos/ics23/go"
+
+	"cosmossdk.io/store/v2/commitment"
+	"cosmossdk.io/store/v2/db"
+)
+
+var _ commitment.Tree = (*Tree)(nil)
+
+// Tree is a simple in-memory implementation of commitment.Tree.
+type Tree struct {
```

Comment:
what is this used for?

Reply:
replicates the usage of memory stores in store/server v1, store v2 usage: https://github.com/cosmos/cosmos-sdk/blob/kocu/runtime-v2/runtime/v2/module.go#L241-L243